src/snarks/groth16/evaluation_from_lagrange.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "include_libsnark.hpp"
+
+#include <map>
+
+/// Given a sequence of powers of x (with some factor) encoded in
+/// GroupT, compute the values of various linear combination of
+/// Lagrange polynomials at x.
+///
+/// TODO: Here, we assume that there may be more Lagrange polynomails
+/// than variables in the original qap, so we use a map to avoid
+/// evaluating those that are unused.  This may not be the optimal approach.
+template<typename ppT, typename GroupT> class evaluation_from_lagrange
```

Comment:
I'd rename this class to something like `evaluator_from_lagrange` to better refer to the notion of "an object". Then, this `evaluator_from_lagrange` could `evaluate_from_langrange_factors`

Reply:
Renamed as suggested.